src/backends/backend.js

```diff
@@ -70,8 +71,12 @@ class Backend {
   currentUser() {
     if (this.user) { return this.user; }
     const stored = this.authStore && this.authStore.retrieve();
-    if (stored) {
-      return Promise.resolve(this.implementation.setUser(stored)).then(() => stored);
+    if (stored && (stored.backend === this.backendName)) {
+      return this.implementation.restoreUser(stored).then((user) => {
+        // return confirmed/new user object instead of stored
+        this.authStore.store({...user, backend: this.backendName});
```

Comment:
Since `this.authStore` defaults to null, we should check for existence first.

Reply:
@erquhart The reason I left that check off here is because we already know `this.authStore` exists -- it had to to get a `stored` value.

Reply:
Do you think it still needs checked?